### PR TITLE
Add `SupportInfo` component for GDPR Privacy Information links

### PR DESCRIPTION
**Pull request by jaswrks**
This cherry picks only the `client/components/support-info/` directory
from https://github.com/Automattic/wp-calypso/pull/23949 so it can be reviewed and merged separately.

### Overview of Changes

- Adds a new `SupportInfo` component. See [README](https://github.com/Automattic/wp-calypso/blob/cb1346aede4d260129652ffb814ce12b9b0320df/client/components/support-info/README.md) in the [`client/components/support-info`](https://github.com/Automattic/wp-calypso/blob/cb1346aede4d260129652ffb814ce12b9b0320df/client/components/support-info) directory. The new component is used to display a support info icon. When clicked, it displays a popover with a description, a learn more link, and a privacy info link.

### CODE REVIEW

File: client/components/support-info/index.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		privacyLink: '',
+	};
+
+	render() {
+		const { text, link, translate } = this.props;
+		let { privacyLink } = this.props;
+
+		if ( ! privacyLink && privacyLink !== false && link ) {
+			privacyLink = link + '#privacy';
+		}
```
Comment (Copons) [inline client/components/support-info/index.jsx]: We could drop the `let` and simplify this into something like this:
```es6
const { privacyLink } = this.props;
const actualPrivacyLink = ( ! privacyLink && privacyLink !== false && link )
  ? link + '#privacy'
  : privacyLink;
```

File: client/components/support-info/style.scss [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+	float: right;
+	margin: 0 0 20px 20px;
+}
+
+.popover {
+	.support-info__privacy {
+		display: block;
+		margin-top: 14px;
+		padding-top: 12px;
+		border-top: 1px solid rgba( $gray, 0.5 );
```
Comment (Copons) [inline client/components/support-info/style.scss]: We should rather use a color variable here:

https://github.com/Automattic/wp-calypso/blob/master/assets/stylesheets/shared/_colors.scss#L10

Anecdotally, in the past I've used `$gray-lighten-30` as a divider border.

**Comment by Copons** [comment]
@jaswrks One thing we should add in a follow up PR is a docs example for this component (which would have been useful to test it 😛).

See for example `InfoPopover` (just to mention a component related to `SupportInfo`): https://github.com/Automattic/wp-calypso/tree/master/client/components/info-popover/docs
https://github.com/Automattic/wp-calypso/blob/master/client/devdocs/design/index.jsx

**Comment by Copons** [comment]
As an aside comment: I love how the lines diff of all the subsequent PRs! 🎉 

Also, in this case, this "base" PR is pretty straightforward and I'm sure will be deployed quickly, but in other cases this might become a blocker for further development.
We've recently tried, successfully I'd dare to say, something similar on a new JP widget.
In that case, the "base" PR was a a more WIP thing, that though we required to develop additional features on top of it.
So we branched the new feature PRs off the "base" PR instead of master.
By doing so is allowing us to work (and test!) on both the base and the other branches at the same time without blockers.

**Comment by kwight** [comment]
Anything else needed to get this to show up in the UI? Adding it to random page (top of the `/comments` page), the element is there, but not displaying the SVG:

screen shot 2018-05-23 at 9 54 52 am

**Comment by Copons** [comment]
@kwight This should be all you need, maybe that location is a bit odd though, with the whole magic going on in our navigation?
I tested on Comments as well, IIRC in `CommentHeader`, and it showed right off the bat.

**Comment by jaswrks** [comment]
@Copons @kwight Thanks so much for the helpful review!

I think I covered all of the above in https://github.com/Automattic/wp-calypso/pull/24985/commits/08a96b508d34c07c0903f1bdd656bf12b0bbb9af. If someone could take one more look I'd appreciate it :-)

- [x] Use `$gray-lighten-30` color code
- [x] Use `const` instead of `let` and be more concise
- [x] Add `docs/example.jsx` and update devdocs index

**Comment by jaswrks** [comment]
> By doing so is allowing us to work (and test!) on both the base and the other branches
Makes sense. I'll do that next time for sure, and once this is merged I'll go ahead and rebase the others.

**Pull request merged**